core/error_api.php

```diff
@@ -183,6 +183,15 @@ function error_handler( $p_type, $p_error, $p_file, $p_line ) {
 		$t_method = DISPLAY_ERROR_HALT;
 	}
 
+	# Special handling for missing language strings, as we do not want to
+	# display information about lang_get() call; instead, we need details
+	# about the actual location of the missing string.
+	if( $p_error == ERROR_LANG_STRING_NOT_FOUND ) {
```

Comment:
Does it work?
`$p_error` is the error message as a string, but `ERROR_LANG_STRING_NOT_FOUND` is the integer id of the message.

Reply:
> Does it work?

Yes

> $p_error is the error message as a string

This is what PHPDoc says for the parameter, but it is not complete; if you look just above, you'll see

```
 * E_USER_* are triggered by us and will contain an error constant in $p_error
 * The others, being system errors, will come with a string in $p_error
```

Reply:
I didn't work for me, so I thought at first sight that it's caused by the comparison.
The real root cause is the recursive call if you are using non-english settings.

Reply:
@dregad not sure if my last note was clear.
You will not get the right file and line if
- your language is set to `auto` and you are running in a non-english environment
- your language is not set to `English`

Reply:
> The real root cause is the recursive call if you are using non-english settings

Can you please describe those settings, and also exactly what you mean by "it didn't work", because I did test with another language and the system behaved as expected.

FTR, the test case is referring to a language string not defined in _strings_english.txt_ (as English is the fall-back language), and obviously also not in the user's current language (English or other).
```php
<?php
include 'core.php';
layout_page_header();
layout_page_begin();
echo lang_get('xxx');
layout_page_end();
```
Master (before)
![image](https://user-images.githubusercontent.com/449891/92323150-9b24c580-f036-11ea-93ed-41bedc7e4884.png)

This PR (after)
![image](https://user-images.githubusercontent.com/449891/92323172-d4f5cc00-f036-11ea-9a7c-cafdecf44b80.png)

Reply:
- In `bug_report_page.php` change `lang_get( 'enter_report_details_title' )` to `lang_get( 'x_enter_report_details_title' )`
- On `My Account` -> `Preferences` change language to `french`
- Click `Rapporter une anomalie`

Result: `Chaîne « x_enter_report_details_title » introuvable. (in '/Applications/MAMP/htdocs/atrol/mastertest/core/lang_api.php' line 304)`

With English settings you get: `String "x_enter_report_details_title" not found. (in '/Applications/MAMP/htdocs/atrol/mastertest/bug_report_page.php' line 253)`

Reply:
Duh. I had it under my own eyes, and did not see it... :blush: Thanks